staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/timing_ratio_histogram_test.go

```diff
@@ -0,0 +1,277 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	compbasemetrics "k8s.io/component-base/metrics"
+	"k8s.io/klog/v2"
+	testclock "k8s.io/utils/clock/testing"
+)
+
+const (
+	ddtRangeCentiPeriods  = 300
+	ddtOffsetCentiPeriods = 50
+	numIterations         = 100
+)
+
+var errMetricNotFound = errors.New("not found")
+
+func TestTimingRatioHistogramVecElementSimple(t *testing.T) {
+	samplesHistName := "tro_test"
+	t0 := time.Now()
+	clk := testclock.NewFakePassiveClock(t0)
+	buckets := []float64{0, 1}
+	vec := NewTestableTimingRatioHistogramVec(clk.Now,
+		&compbasemetrics.TimingHistogramOpts{Name: samplesHistName, Buckets: buckets},
+		"alabel",
+	)
+	toRegister := vec.metrics()
+	registry := compbasemetrics.NewKubeRegistry()
+	for _, reg := range toRegister {
+		registry.MustRegister(reg)
+	}
+	tro, err := vec.NewForLabelValuesChecked(0, 1, []string{"avalue"})
+	if err != nil {
+		t.Error(err)
+	}
+	exerciseTimingRatioHistogram(t, samplesHistName, t0, clk, registry, tro)
+}
+
+func TestTimingRatioHistogramVecElementSafeEarly(t *testing.T) {
+	samplesHistName := "tro_test"
+	t0 := time.Now()
+	clk := testclock.NewFakePassiveClock(t0)
+	buckets := []float64{0, 1}
+	vec := NewTestableTimingRatioHistogramVec(clk.Now,
+		&compbasemetrics.TimingHistogramOpts{Name: samplesHistName, Buckets: buckets},
+		"alabel",
+	)
+	tro := vec.NewForLabelValuesSafe(0, 1, []string{"avalue"})
+	toRegister := vec.metrics()
+	registry := compbasemetrics.NewKubeRegistry()
+	for _, reg := range toRegister {
+		registry.MustRegister(reg)
+	}
+	exerciseTimingRatioHistogram(t, samplesHistName, t0, clk, registry, tro)
+}
+
+func TestTimingRatioHistogramVecElementSafeLate(t *testing.T) {
+	samplesHistName := "tro_test"
+	t0 := time.Now()
+	clk := testclock.NewFakePassiveClock(t0)
+	buckets := []float64{0, 1}
+	vec := NewTestableTimingRatioHistogramVec(clk.Now,
+		&compbasemetrics.TimingHistogramOpts{Name: samplesHistName, Buckets: buckets},
+		"alabel",
+	)
+	toRegister := vec.metrics()
+	registry := compbasemetrics.NewKubeRegistry()
+	for _, reg := range toRegister {
+		registry.MustRegister(reg)
+	}
+	tro := vec.NewForLabelValuesSafe(0, 1, []string{"avalue"})
+	exerciseTimingRatioHistogram(t, samplesHistName, t0, clk, registry, tro)
+}
+
+/* exerciseTimingRatioHistogram does a rough behavioral test of a
```

Comment:
nit: let's use `// in each line` comment style for consistency with the whole codebase

Reply:
done